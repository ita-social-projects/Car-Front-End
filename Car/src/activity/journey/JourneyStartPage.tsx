import React, { useCallback, useContext, useEffect, useState } from "react";
import { RefreshControl, ScrollView, Text, TouchableOpacity, View } from "react-native";
import JourneyService from "../../../api-service/journey-service/JourneyService";
import Journey from "../../../models/Journey";
import AuthContext from "../../components/auth/AuthContext";
import JourneyCardList from "../../components/journey-card/JourneyCardList";
import JourneyStartPageStyle from "./JourneyStartPageStyle";
import TouchableNavigationBlock from "../../components/touchable-navigation-block/TouchableNavigationBlock";
import {
    FIRST_ELEMENT_INDEX,
    SECOND_ELEMENT_INDEX,
    THIRD_ELEMENT_INDEX,
    FOURTH_ELEMENT_INDEX,
    EMPTY_COLLECTION_LENGTH
} from "../../constants/Constants";
import DM from "../../components/styles/DM";

const JourneyStartPage = (props: any) => {

    const activeButtonStyle = {
        backgroundColor: DM("#000000"),
        color: DM("#FFFFFF")
    };

    const unactiveButtonStyle = {
        backgroundColor: DM("#FFFFFF"),
        color: DM("#000000")
    };

    const [selectedIndex, setSelectedIndex] = useState(FIRST_ELEMENT_INDEX);
    const [allButtonStyle, setAllButtonStyle] = useState(activeButtonStyle);
    const [pastButtonStyle, setPastButtonStyle] = useState(unactiveButtonStyle);
    const [upcomingButtonStyle, setUpcomingButtonStyle] = useState(unactiveButtonStyle);
    const [scheduledButtonStyle, setScheduledButtonStyle] = useState(unactiveButtonStyle);

    const { user } = useContext(AuthContext);
    const [pastJourneys, setPastJourneys] = useState<Array<Journey>>([]);
    const [upcomingJourneys, setUpcomingJourneys] = useState<Array<Journey>>(
        []
    );
    const [scheduledJourneys, setScheduledJourneys] = useState<Array<Journey>>(
        []
    );

    const [refreshing, setRefreshing] = useState(false);

    const onRefresh = useCallback(() => {
        setRefreshing(true);
        loadJourneys();
    }, []);

    const loadJourneys = () => {
        JourneyService.getUpcomingJourneys(Number(user?.id)).then((res) =>
            setUpcomingJourneys(res.data)
        ).then(() => setRefreshing(false));

        JourneyService.getPastJourneys(Number(user?.id)).then((res1) =>
            setPastJourneys(res1.data)
        ).then(() => setRefreshing(false));

        JourneyService.getScheduledJourneys(Number(user?.id)).then((res2) =>
            setScheduledJourneys(res2.data)
        ).then(() => setRefreshing(false));
    };

    useEffect(() => {
        return props.navigation.addListener("focus", loadJourneys);
    }, [props.navigation]);

    return (
        <ScrollView
            refreshControl={
                <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
            }
            style={[JourneyStartPageStyle.page, { backgroundColor: DM("white") }]}>
            <View style={JourneyStartPageStyle.touchableNavigationBlocks}>
                <TouchableNavigationBlock
                    navigation={props.navigation}
                    navigationName="Search Journey"
                    blockImage={require("../../../assets/images/journey/bermuda-searching.png")}
                    blockName="Search for a Journey"
                    from="#A5C500"
                    to="#00A977"
                    reverse={false}
                    width={150}
                    height={140}
                />
                <TouchableNavigationBlock
                    navigation={props.navigation}
                    navigationName="Create Journey"
                    blockImage={require("../../../assets/images/journey/bermuda-delivery-car-service.png")}
                    blockName="Create a Journey"
                    from="#00A3CF"
                    to="#5552A0"
                    reverse={true}
                    width={210}
                    height={140}
                />
            </View>
            <View style={JourneyStartPageStyle.manageJourneysContainer}>
                <Text style={[JourneyStartPageStyle.manageJourneysText, { color: DM("black") }]}>
                    Manage journeys
                </Text>
            </View>
            <View style={JourneyStartPageStyle.segmentControlContainer}>
                <TouchableOpacity
                    activeOpacity={1}
                    style={[JourneyStartPageStyle.allJourneys, allButtonStyle, { borderColor: DM("black") }]}
                    onPress={() => {
                        setSelectedIndex(FIRST_ELEMENT_INDEX);
                        setAllButtonStyle(activeButtonStyle);
                        setPastButtonStyle(unactiveButtonStyle);
                        setUpcomingButtonStyle(unactiveButtonStyle);
                        setScheduledButtonStyle(unactiveButtonStyle);
                    }}
                >
                    <Text
                        style={[
                            JourneyStartPageStyle.buttonText,
                            allButtonStyle
                        ]}
                    >
                        All
                    </Text>
                </TouchableOpacity>
                <TouchableOpacity
                    activeOpacity={1}
                    style={[
                        JourneyStartPageStyle.pastJourneys,
                        pastButtonStyle,
                        { borderColor: DM("black") }
                    ]}
                    onPress={() => {
                        setSelectedIndex(SECOND_ELEMENT_INDEX);
                        setAllButtonStyle(unactiveButtonStyle);
                        setPastButtonStyle(activeButtonStyle);
                        setUpcomingButtonStyle(unactiveButtonStyle);
                        setScheduledButtonStyle(unactiveButtonStyle);
                    }}
                >
                    <Text
                        style={[
                            JourneyStartPageStyle.buttonText,
                            pastButtonStyle
                        ]}
                    >
                        Past
                    </Text>
                </TouchableOpacity>
                <TouchableOpacity
                    activeOpacity={1}
                    style={[
                        JourneyStartPageStyle.upcomingJourneys,
                        upcomingButtonStyle,
                        { borderColor: DM("black") }
                    ]}
                    onPress={() => {
                        setSelectedIndex(THIRD_ELEMENT_INDEX);
                        setAllButtonStyle(unactiveButtonStyle);
                        setPastButtonStyle(unactiveButtonStyle);
                        setUpcomingButtonStyle(activeButtonStyle);
                        setScheduledButtonStyle(unactiveButtonStyle);
                    }}
                >
                    <Text
                        style={[
                            JourneyStartPageStyle.buttonText,
                            upcomingButtonStyle
                        ]}
                    >
                        Upcoming
                    </Text>
                </TouchableOpacity>
                <TouchableOpacity
                    activeOpacity={1}
                    style={[
                        JourneyStartPageStyle.scheduledJourneys,
                        scheduledButtonStyle,
                        { borderColor: DM("black") }
                    ]}
                    onPress={() => {
                        setSelectedIndex(FOURTH_ELEMENT_INDEX);
                        setAllButtonStyle(unactiveButtonStyle);
                        setPastButtonStyle(unactiveButtonStyle);
                        setUpcomingButtonStyle(unactiveButtonStyle);
                        setScheduledButtonStyle(activeButtonStyle);
                    }}
                >
                    <Text
                        style={[
                            JourneyStartPageStyle.buttonText,
                            scheduledButtonStyle
                        ]}
                    >
                        Scheduled
                    </Text>
                </TouchableOpacity>
            </View>

            {selectedIndex === FIRST_ELEMENT_INDEX && (
                <View style={JourneyStartPageStyle.tabStyle}>
                    {upcomingJourneys.length > EMPTY_COLLECTION_LENGTH && (
                        <Text style={[JourneyStartPageStyle.tabTextStyle, { color: DM("black") }]}>
                            Upcoming
                        </Text>
                    )}
                    {<JourneyCardList journey={upcomingJourneys} />}

                    {pastJourneys.length > EMPTY_COLLECTION_LENGTH && (
                        <Text style={[JourneyStartPageStyle.tabTextStyle, { color: DM("black") }]}>
                            Past
                        </Text>
                    )}
                    {<JourneyCardList journey={pastJourneys} />}

                    {scheduledJourneys.length > EMPTY_COLLECTION_LENGTH && (
                        <Text style={[JourneyStartPageStyle.tabTextStyle, { color: DM("black") }]}>
                            Scheduled
                        </Text>
                    )}
                    {<JourneyCardList journey={scheduledJourneys} />}
                </View>
            )}
            {selectedIndex === SECOND_ELEMENT_INDEX && (
                <View style={JourneyStartPageStyle.tabStyle}>
                    {<JourneyCardList journey={pastJourneys} />}
                </View>
            )}
            {selectedIndex === THIRD_ELEMENT_INDEX && (
                <View style={JourneyStartPageStyle.tabStyle}>
                    {<JourneyCardList journey={upcomingJourneys} />}
                </View>
            )}
            {selectedIndex === FOURTH_ELEMENT_INDEX && (
                <View style={JourneyStartPageStyle.tabStyle}>
                    {<JourneyCardList journey={scheduledJourneys} />}
                </View>
            )}
        </ScrollView>
    );
};

export default JourneyStartPage;
