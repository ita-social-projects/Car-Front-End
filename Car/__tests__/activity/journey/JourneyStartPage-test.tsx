import React from "react";
import shallowRenderer from "react-test-renderer/shallow";
import JourneyStartPage from "../../../src/activity/journey/JourneyStartPage";

const renderer = shallowRenderer.createRenderer();

test("renders correctly", () =>
  expect(renderer.render(<JourneyStartPage />)).toMatchInlineSnapshot(`
    <ScrollView
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      style={
        Object {
          "backgroundColor": "white",
          "paddingHorizontal": 15,
        }
      }
    >
      <View
        style={
          Object {
            "paddingTop": 16,
          }
        }
      >
        <TouchableNavigationBlock
          blockImage={
            Object {
              "testUri": "../../../assets/images/journey/bermuda-searching.png",
            }
          }
          blockName="Search for a Journey"
          from="#A5C500"
          height={140}
          navigationName="Search Journey"
          reverse={false}
          to="#00A977"
          width={150}
        />
        <TouchableNavigationBlock
          blockImage={
            Object {
              "testUri": "../../../assets/images/journey/bermuda-delivery-car-service.png",
            }
          }
          blockName="Create a Journey"
          from="#00A3CF"
          height={140}
          navigationName="Create Journey"
          reverse={true}
          to="#5552A0"
          width={210}
        />
      </View>
      <View
        style={
          Object {
            "paddingTop": 40,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Proxima Nova",
              "fontSize": 20,
              "fontWeight": "700",
              "textTransform": "uppercase",
            }
          }
        >
          Manage journeys
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 56,
            "paddingTop": 24,
          }
        }
      >
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "black",
                "borderWidth": 2,
                "justifyContent": "center",
                "width": 54,
              },
              Object {
                "backgroundColor": "#000000",
                "color": "#FFFFFF",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova",
                  "fontWeight": "700",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#000000",
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            All
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "black",
                "borderWidth": 2,
                "justifyContent": "center",
                "width": 62,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "#000000",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova",
                  "fontWeight": "700",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#000000",
                },
              ]
            }
          >
            Past
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "black",
                "borderWidth": 2,
                "justifyContent": "center",
                "width": 98,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "#000000",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova",
                  "fontWeight": "700",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#000000",
                },
              ]
            }
          >
            Upcoming
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "black",
                "borderWidth": 2,
                "justifyContent": "center",
                "width": 103,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "#000000",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova",
                  "fontWeight": "700",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#000000",
                },
              ]
            }
          >
            Scheduled
          </Text>
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "paddingTop": 24,
          }
        }
      >
        <JourneyCardList
          journey={Array []}
        />
        <JourneyCardList
          journey={Array []}
        />
        <JourneyCardList
          journey={Array []}
        />
      </View>
    </ScrollView>
  `));
