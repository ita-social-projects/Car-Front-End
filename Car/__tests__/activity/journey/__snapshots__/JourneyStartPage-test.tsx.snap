// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: 
    <ScrollView
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "paddingHorizontal": 7,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "paddingTop": 16,
          }
        }
      >
        <TouchableNavigationBlock
          blockImage={
            Object {
              "testUri": "../../../assets/images/journey/bermuda-searching.png",
            }
          }
          blockName="Find a Ride"
          from="#A5C500"
          height={140}
          navigation={
            Object {
              "addListener": [Function],
            }
          }
          navigationName="Search Journey"
          reverse={false}
          to="#00A997"
          width={150}
        />
        <TouchableNavigationBlock
          blockImage={
            Object {
              "testUri": "../../../assets/images/journey/bermuda-delivery-car-service.png",
            }
          }
          blockName="Add a ride"
          from="#00A3CF"
          height={140}
          navigation={
            Object {
              "addListener": [Function],
            }
          }
          navigationName="Create Journey"
          reverse={true}
          to="#5552A0"
          width={200}
        />
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 7,
          }
        }
      >
        <View
          style={
            Object {
              "paddingTop": 40,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Black",
                  "fontSize": 25,
                  "fontWeight": "700",
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#0B171B",
                },
              ]
            }
          >
            MANAGE RIDES
          </Text>
        </View>
        <ScrollView
          contentContainerStyle={
            Object {
              "flexGrow": 10,
            }
          }
          horizontal={true}
          pagingEnabled={true}
          showsHorizontalScrollIndicator={false}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 56,
                "paddingTop": 24,
                "width": 450,
              }
            }
          >
            <ForwardRef
              activeOpacity={1}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "width": "13%",
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderBottomWidth": 2,
                    "color": "#0B171B",
                  },
                  Object {
                    "borderColor": "#0B171B",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Proxima Nova Extrabold",
                      "fontWeight": "300",
                      "textTransform": "uppercase",
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "color": "#0B171B",
                    },
                  ]
                }
              >
                All
              </Text>
            </ForwardRef>
            <ForwardRef
              activeOpacity={1}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "width": "13%",
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderBottomWidth": 0,
                    "color": "#414045",
                  },
                  Object {
                    "borderColor": "#0B171B",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Proxima Nova Extrabold",
                      "fontWeight": "300",
                      "textTransform": "uppercase",
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "color": "#414045",
                    },
                  ]
                }
              >
                Past
              </Text>
            </ForwardRef>
            <ForwardRef
              activeOpacity={1}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "width": "24%",
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderBottomWidth": 0,
                    "color": "#414045",
                  },
                  Object {
                    "borderColor": "#0B171B",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Proxima Nova Extrabold",
                      "fontWeight": "300",
                      "textTransform": "uppercase",
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "color": "#414045",
                    },
                  ]
                }
              >
                Upcoming
              </Text>
            </ForwardRef>
            <ForwardRef
              activeOpacity={1}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "width": "20%",
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderBottomWidth": 0,
                    "color": "#414045",
                  },
                  Object {
                    "borderColor": "#0B171B",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Proxima Nova Extrabold",
                      "fontWeight": "300",
                      "textTransform": "uppercase",
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "color": "#414045",
                    },
                  ]
                }
              >
                Regular
              </Text>
            </ForwardRef>
          </View>
        </ScrollView>
        <View
          style={
            Object {
              "paddingTop": 24,
            }
          }
        >
          <JourneyCardList
            ascending={true}
            journey={Array []}
          />
          <JourneyCardList
            isPast={true}
            journey={Array []}
          />
          <JourneyCardList
            journey={Array []}
          />
        </View>
      </View>
    </ScrollView>
'  || 
<ScrollView
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingHorizontal": 7,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingTop": 16,
      }
    }
  >
    <TouchableNavigationBlock
      blockImage={
        Object {
          "testUri": "../../../assets/images/journey/bermuda-searching.png",
        }
      }
      blockName="Find a Ride"
      from="#A5C500"
      height={108}
      navigation={
        Object {
          "addListener": [Function],
        }
      }
      navigationName="Search Journey"
      reverse={false}
      to="#00A997"
      width={140}
    />
    <TouchableNavigationBlock
      blockImage={
        Object {
          "testUri": "../../../assets/images/journey/bermuda-delivery-car-service.png",
        }
      }
      blockName="Add a ride"
      from="#00A3CF"
      height={100}
      navigation={
        Object {
          "addListener": [Function],
        }
      }
      navigationName="Create Journey"
      reverse={true}
      to="#5552A0"
      width={150}
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 7,
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 40,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Proxima Nova Black",
              "fontSize": 25,
              "fontWeight": "700",
              "textTransform": "uppercase",
            },
            Object {
              "color": "#0B171B",
            },
          ]
        }
      >
        MANAGE RIDES
      </Text>
    </View>
    <ScrollView
      contentContainerStyle={
        Object {
          "flexGrow": 10,
        }
      }
      horizontal={true}
      pagingEnabled={true}
      showsHorizontalScrollIndicator={false}
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 56,
            "paddingTop": 24,
            "width": 450,
          }
        }
      >
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "width": "13%",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomWidth": 2,
                "color": "#0B171B",
              },
              Object {
                "borderColor": "#0B171B",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Extrabold",
                  "fontWeight": "300",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#0B171B",
                },
              ]
            }
          >
            All
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "width": "13%",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomWidth": 0,
                "color": "#414045",
              },
              Object {
                "borderColor": "#0B171B",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Extrabold",
                  "fontWeight": "300",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#414045",
                },
              ]
            }
          >
            Past
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "width": "24%",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomWidth": 0,
                "color": "#414045",
              },
              Object {
                "borderColor": "#0B171B",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Extrabold",
                  "fontWeight": "300",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#414045",
                },
              ]
            }
          >
            Upcoming
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "width": "20%",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomWidth": 0,
                "color": "#414045",
              },
              Object {
                "borderColor": "#0B171B",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Extrabold",
                  "fontWeight": "300",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#414045",
                },
              ]
            }
          >
            Regular
          </Text>
        </ForwardRef>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "paddingTop": 24,
        }
      }
    >
      <JourneyCardList
        ascending={true}
        journey={Array []}
      />
      <JourneyCardList
        isPast={true}
        journey={Array []}
      />
      <JourneyCardList
        journey={Array []}
      />
    </View>
  </View>
</ScrollView>
 1`] = `
<ScrollView
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingHorizontal": 7,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingTop": 16,
      }
    }
  >
    <TouchableNavigationBlock
      blockImage={
        Object {
          "testUri": "../../../assets/images/journey/bermuda-searching.png",
        }
      }
      blockName="Find a Ride"
      from="#A5C500"
      height={140}
      navigation={
        Object {
          "addListener": [Function],
        }
      }
      navigationName="Search Journey"
      reverse={false}
      to="#00A997"
      width={150}
    />
    <TouchableNavigationBlock
      blockImage={
        Object {
          "testUri": "../../../assets/images/journey/bermuda-delivery-car-service.png",
        }
      }
      blockName="Add a ride"
      from="#00A3CF"
      height={140}
      navigation={
        Object {
          "addListener": [Function],
        }
      }
      navigationName="Create Journey"
      reverse={true}
      to="#5552A0"
      width={200}
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 7,
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 40,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Proxima Nova Black",
              "fontSize": 25,
              "fontWeight": "700",
              "textTransform": "uppercase",
            },
            Object {
              "color": "#0B171B",
            },
          ]
        }
      >
        MANAGE RIDES
      </Text>
    </View>
    <ScrollView
      contentContainerStyle={
        Object {
          "flexGrow": 10,
        }
      }
      horizontal={true}
      pagingEnabled={true}
      showsHorizontalScrollIndicator={false}
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 56,
            "paddingTop": 24,
            "width": 450,
          }
        }
      >
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "width": "13%",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomWidth": 2,
                "color": "#0B171B",
              },
              Object {
                "borderColor": "#0B171B",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Extrabold",
                  "fontWeight": "300",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#0B171B",
                },
              ]
            }
          >
            All
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "width": "13%",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomWidth": 0,
                "color": "#414045",
              },
              Object {
                "borderColor": "#0B171B",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Extrabold",
                  "fontWeight": "300",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#414045",
                },
              ]
            }
          >
            Past
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "width": "24%",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomWidth": 0,
                "color": "#414045",
              },
              Object {
                "borderColor": "#0B171B",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Extrabold",
                  "fontWeight": "300",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#414045",
                },
              ]
            }
          >
            Upcoming
          </Text>
        </ForwardRef>
        <ForwardRef
          activeOpacity={1}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "width": "20%",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomWidth": 0,
                "color": "#414045",
              },
              Object {
                "borderColor": "#0B171B",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Proxima Nova Extrabold",
                  "fontWeight": "300",
                  "textTransform": "uppercase",
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "color": "#414045",
                },
              ]
            }
          >
            Regular
          </Text>
        </ForwardRef>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "paddingTop": 24,
        }
      }
    >
      <JourneyCardList
        ascending={true}
        journey={Array []}
      />
      <JourneyCardList
        isPast={true}
        journey={Array []}
      />
      <JourneyCardList
        journey={Array []}
      />
    </View>
  </View>
</ScrollView>
`;
